pennylane/operation.py

```diff
@@ -713,6 +715,44 @@ def __rmatmul__(self, other):
 
     __imatmul__ = __matmul__
 
+    @property
+    def eigvals(self):
+        """Return the eigenvalues of the specified tensor product observable by
+        using pre-stored eigenvalues for standard observables where possible.
+
+        Returns:
+            array: array containing the eigenvalues of the tensor product observable
+        """
+        standard_observables = {"PauliX", "PauliY", "PauliZ", "Hadamard"}
+
+        # observable should be Z^{\otimes n}
+        self._eigvals = pauli_eigs(len(self.wires))
+
+        # check if there are any non-standard observables (such as Identity)
+        if set(self.obs) - standard_observables:
+            # Tensor product of observables contains a mixture
+            # of standard and non-standard observables
+            self._eigvals = np.array([1])
+            for k, g in itertools.groupby(zip(self.obs, self.wires,\
+                self.parameters), lambda x: x[0].name in standard_observables):
+                if k:
+                    # Subgroup g contains only standard observables.
+                    # Determine the size of the subgroup, by transposing
+                    # the list, flattening it, and determining the length.
+                    n = len([w for sublist in list(zip(*g))[1] for w in sublist])
+                    self._eigvals = np.kron(self._eigvals, pauli_eigs(n))
+                else:
+                    # Subgroup g contains only non-standard observables.
+                    for ns_obs in g:
+                        # loop through all non-standard observables
+                        if ns_obs[0].name == "Hermitian":
+                            # Hermitian observable has pre-computed eigenvalues
+                            operator = ns_obs[2]
+                            herm_eigs = qml.Hermitian.eigvals(np.array(operator))
+                            self._eigvals = np.kron(self._eigvals, herm_eigs)
+
+        return self._eigvals
```

Comment:
Really nice! However, should there be a
```python
if self._eigvals is not None:
    return self._eigvals
```
at the very beginning? Otherwise it looks like this will run with every call